plugins/modules/virtual_disk.py

```diff
@@ -0,0 +1,197 @@
+    virtual_disk_obj: Optional[VirtualDisk],
+) -> Tuple[bool, Union[TypedVirtualDiskToAnsible, None], TypedDiff]:
+    before = None
+    after = None
+    if virtual_disk_obj:
+        return False, after, dict(before=before, after=after)
+    else:
+        file_content, file_size = read_disk_file(module)
+        VirtualDisk.send_upload_request(
+            rest_client, file_content, file_size, module.params["file_name"]
+        )
+        # No wait_task needed; Upload not returning task ID.
+        updated_disk = VirtualDisk.get_by_name(
+            rest_client, name=module.params["file_name"], must_exist=True
+        )
+        after = updated_disk.to_ansible() if updated_disk else None
+        return is_changed(before, after), after, dict(before=before, after=after)
+
+
+def ensure_absent(
+    module: AnsibleModule,
+    rest_client: RestClient,
+    virtual_disk_obj: Optional[VirtualDisk],
+) -> Tuple[bool, Union[TypedVirtualDiskToAnsible, None], TypedDiff]:
+    before = None
+    after = None
+    if not virtual_disk_obj:
+        return False, after, dict(before=before, after=after)
+    else:
+        before = virtual_disk_obj.to_ansible()
+        virtual_disk_obj.send_delete_request(rest_client)
+        # No wait_task needed; Upload not returning task ID.
+        updated_disk = VirtualDisk.get_by_name(
+            rest_client, name=module.params["file_name"]
+        )
+        after = updated_disk.to_ansible() if updated_disk else None
+        return is_changed(before, after), after, dict(before=before, after=after)
+
+
+# Virtual disk can only be created or deleted; No update actions available.
+def run(
+    module: AnsibleModule, rest_client: RestClient
+) -> Tuple[bool, Union[TypedVirtualDiskToAnsible, None], TypedDiff]:
+    virtual_disk_obj = VirtualDisk.get_by_name(
+        rest_client, name=module.params["file_name"]
+    )
+    if module.params["state"] == State.present:
+        return ensure_present(module, rest_client, virtual_disk_obj)
+    return ensure_absent(module, rest_client, virtual_disk_obj)
+
+
+def main() -> None:
+    module = AnsibleModule(
+        supports_check_mode=False,
+        argument_spec=dict(
+            arguments.get_spec("cluster_instance"),
+            state=dict(
+                type="str",
+                choices=[
+                    "present",
+                    "absent",
+                ],
+                required=True,
+            ),
+            file_location=dict(
+                type="str",
+            ),
+            file_name=dict(
+                type="str",
+                required=True,
+            ),
+        ),
+        required_if=[("state", "present", ("file_location",), False)],
+    )
+
+    try:
+        client = Client.get_client(module.params["cluster_instance"])
+        rest_client = RestClient(client)
+        hcversion = HyperCoreVersion(rest_client)
+        if not hcversion.verify(HYPERCORE_VERSION_REQUIREMENTS):
```

Comment:
Maybe add a `version_check()` function instead of this.
Make this a one function call.